### ISSUE TITLE
Does MiNode always advertise the correct port?

### ISSUE THREAD

**Issue by kseistrup** [TheKysek/MiNode#9]
TL;DR: Does MiNode incorrectly advertise the portnumber(s) of one or more of its outgoing connections as its listening address?

I have a TCP logger running on the same box as MiNode and when someone attempts a `SYN` on the net device a line like

```
Aug  4 16:15:43 example tcplog[123]: port 8444 (bitmessage) request from node.example.com[1.2.3.4]
```

will be written to syslog.

MiNode and PyBitmessage in general both listen on port 8444 so that when net connections are listed with `netstat -anp` incoming connection will have port 8444 in the left hand column and a port > 1024 (but not 8444) in the right hand column, whereas outgoing connections will have port 8444 in the right hand column and a port > 1024 (but not 8444) in the left hand column.

Now, what puzzles me about the box where I'm running MiNode is that I see a lot of “`… port NNNNN (unknown) request from …`” lines from `tcplog`, where NNNNN > 1024 (but not 8444). And when I look up NNNNN with `netstat(8)` the port often happens to be one of MiNode's outgoing connections (sometimes the port is not listed in `netstat(8)`, but it _never_ belongs to another listening program than MiNode). I _never_ see this pattern on a box that runs PyBitmessage.

AFAIK, all connections in MiNode (and PyBitmessage) are unidirectional, so this begs the question: Does MiNode incorrectly advertise the portnumber(s) of one or more of its outgoing connections as its listening address?

**Comment by kseistrup** [comment]
In addition to what is mentioned above, I just learned that MiNode itself also seems to attempt connecting to lots of non-standard ports, e.g.:

```
[2016-08-04 15:59:15,609] [WARNING] Connection to a.b.c.d:53656 failed
```

and I find it hard to believe that so many nodes are running on ports other than 8444, so I suspect that the problem affects both incoming and outgoing connections.

**Comment by kseistrup** [comment]
I've stopped my MiNode until this bug is resolved. MiNode “pollutes” the peer address pool with faulty address:port combinations — that can't be good for anyone.

**Comment by TheKysek** [comment]
I did some logging of advertised addresses and they seemed fine.
The code also looks fine to me.
I will still keep an eye on it.

```
if not self.server:
    # If it is outgoing connection, advertise address of the peer I am connected to and add it to node_pool.
    shared.address_advertise_queue.put(structure.NetAddr(version.services, self.host, self.port))
    shared.node_pool.add((self.host, self.port))
# Always advertise IP remote node told me I have with my listening port.
shared.address_advertise_queue.put(structure.NetAddr(shared.services, version.host, shared.listening_port))
```

**Comment by kseistrup** [comment]
Can you please confirm that your own MiNode instance attempts to establish outgoing connections to a lot of non-standard ports (port > 1024 AND port != 8444) and fails?

**Comment by TheKysek** [comment]
Yes, there are some failing connections to non-standard ports. It looks like there are faulty addresses being advertised in the network and they are landing in unchecked_node_pool. Still, it doesn't seem like MiNode is doing that.

```
[2016-08-05 17:53:46,468] [DEBUG] Received advertisement of 91.67.15.67:8444
[2016-08-05 17:53:46,468] [DEBUG] Received advertisement of 1.234.89.222:8444
[2016-08-05 17:53:46,468] [DEBUG] Received advertisement of 141.237.80.192:49488 <<
[2016-08-05 17:53:46,468] [DEBUG] Received advertisement of 80.108.53.142:35407 <<
[2016-08-05 17:53:46,468] [DEBUG] Received advertisement of 66.18.39.225:49535 <<
[2016-08-05 17:53:46,468] [DEBUG] Received advertisement of 191.5.217.179:8444
[2016-08-05 17:53:46,468] [DEBUG] Received advertisement of 2002:d588:4c25::d588:4c25:8444
```

**Comment by kseistrup** [comment]
I have to agree that it doesn't seem like MiNode is advertising rogue addresses, judging from the code. So let's leave it at that. Thanks for your help.
